### PR TITLE
feat: add metadata filter for all UI tables using clickhouse

### PR DESCRIPTION
**Pull request by maxdeichmann**
> [!IMPORTANT]
> Add `StringObjectFilter` for Clickhouse map filtering and enhance query logging with query IDs in development mode.
> 
>   - **Behavior**:
>     - Add `StringObjectFilter` class in `clickhouse-filter.ts` for key-value pair filtering in Clickhouse maps.
>     - Integrate `StringObjectFilter` in `createFilterFromFilterState()` in `factory.ts`.
>     - Update `queryClickhouse()` in `clickhouse.ts` to log query IDs in development mode.
>   - **Logging**:
>     - Log query ID with query text in `queryClickhouse()` when in development mode.
>   - **Error Handling**:
>     - Log error for invalid filter type in `createFilterFromFilterState()` in `factory.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for 88d197ec51c29e1ccabf310791075e996aa7183a. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: packages/shared/src/server/repositories/clickhouse.ts [hunk 1]
```diff
@@ -34,16 +34,17 @@ export async function queryClickhouse<T>(opts: {
...
 
     const res = await clickhouseClient.query({
       query: opts.query,
       format: "JSONEachRow",
       query_params: opts.params,
     });
 
+    if (env.NODE_ENV === "development") {
+      logger.info(`clickhouse:query ${res.query_id} ${opts.query}`);
+    }
```
[the remark this answers is not in the document]
Comment (maxdeichmann) [reply packages/shared/src/server/repositories/clickhouse.ts:46]: I would like to log our the queryId as well to find it in the clickhouse logs.

File: packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts [hunk 2]
```diff
@@ -169,6 +169,66 @@ export class StringOptionsFilter implements Filter {
...
+  protected operator: (typeof filterOperators)["stringObject"][number];
+  protected tablePrefix?: string;
+
+  constructor(opts: {
+    clickhouseTable: string;
+    field: string;
+    operator: (typeof filterOperators)["stringObject"][number];
+    key: string;
+    value: string;
+    tablePrefix?: string;
```
[the remark this answers is not in the document]
Comment (maxdeichmann) [reply packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:188]: correct: 

```
return new StringObjectFilter({
          clickhouseTable: column.clickhouseTableName,
          field: column.clickhouseSelect,
          operator: frontEndFilter.operator,
          key: frontEndFilter.key,
          value: frontEndFilter.value,
          tablePrefix: column.queryPrefix,
        });
```

[the remark this answers is not in the document]
Comment (maxdeichmann) [reply packages/shared/src/server/queries/clickhouse-filter/clickhouse-filter.ts:188]: we search for the column which is the one defined in code.

File: packages/shared/src/server/repositories/clickhouse.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxdeichmann) [reply packages/shared/src/server/repositories/clickhouse.ts:44]: I moved the comment

**Review by Steffen911 — APPROVED** [review]

**Pull request merged**